### PR TITLE
Enable global load balancer and extra SANs

### PR DESCRIPTION
**Pull request by bgeesaman**
- Move from regional to global load balancer
- Adds support for custom DNS names in the TLS certs

The global LB takes about 3m more on average to provision and become healthy.

### CODE REVIEW

**Comment by sethvargo** [comment]
Hey @bgeesaman 

Thanks for this. I'm going to hold off until after KubeCon to test and review this. I just don't have the cycles at the moment, sorry 😄

**Comment by bgeesaman** [comment]
Of course!  I just had some cycles on the plane ride out here.

File: k8s/vault.yaml [hunk 1]
```diff
@@ -1,18 +1,32 @@
+apiVersion: extensions/v1beta1
```
Comment (chrislovecnm) [inline k8s/vault.yaml:1]: Dumb question.  Are we still on beta on the Ingress API?

Comment (chrislovecnm) [inline k8s/vault.yaml:1]: Please include documentation to the ingress API. 

@sethvargo do you have Apache headers in your manifests?

File: terraform/yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (chrislovecnm) [inline terraform/yml]: a file named "yml"? I am confused

**Review by chrislovecnm — COMMENTED** [review]
Questions for you

File: k8s/vault.yaml (hunk 1 above)
Comment (bgeesaman) [reply k8s/vault.yaml:1]: Followed https://cloud.google.com/kubernetes-engine/docs/tutorials/http-balancer#step_3_create_an_ingress_resource for this component.

File: terraform/yml (hunk 2 above)
Comment (bgeesaman) [reply terraform/yml]: Thanks.  Removed.

**Comment by bgeesaman** [comment]
Reviewing this further, I think it might make the most sense to generate a separate keypair with just the public IP and optional custom hostnames for use by the GCLB front end.  The existing cert can remain as-is.  This would prevent the external exposure of a cert with SANs of localhost and 127.0.0.1

**Pull request closed without merging**